macosx/NSMutableArray+Additions.swift

```diff
@@ -0,0 +1,80 @@
+// This file Copyright © 2011-2023 Transmission authors and contributors.
+// It may be used under the MIT (SPDX: MIT) license.
+// License text can be found in the licenses/ folder.
+
+import Foundation
+
+public extension NSMutableArray {
+    /*
+     Note: This assumes Apple implemented this as an array under the hood.
+     If the underlying data structure was a linked-list, for example, then this might be less
+     efficient than simply removing the object and re-adding it.
+     */
+    @objc
+    func moveObjectAtIndex(_ fromIndex: Int, toIndex: Int) {
+        if fromIndex == toIndex {
+            return
+        }
+
+        // shift objects - more efficient than simply removing the object and re-inserting the object
+        let object = self[fromIndex]
+        if fromIndex < toIndex {
+            var i = fromIndex
+            while i < toIndex {
+                self[i] = self[i + 1]
+                i += 1
+            }
+        } else {
+            var i = fromIndex
+            while i > toIndex {
+                self[i] = self[i - 1]
+                i -= 1
+            }
+        }
+        self[toIndex] = object
+    }
+}
+
+extension Array {
+    /*
+     Note: This assumes Swift implemented this as an array under the hood.
+     If the underlying data structure was a linked-list, for example, then this might be less
+     efficient than simply removing the object and re-adding it.
+     */
+    /*
+     Benchmarking on macOS 13.1, a move fromIndex:1000 toIndex:0
+     - Fastest: simply removing the object and re-adding it
```

Comment:
If it's the fastest, why not keep it simple?

Reply:
I work a year ago for a whole day on this small file, comparing many implementations. Probably the difference was negligible, but also I thought that this api is called only in response to reordering visible elements of the UI, and that you hardly display more than 50 items on the screen at the same time. And for small moves, shifting is probably the best solution.

Maybe the ideal solution is to compare abs(fromIndex - toIndex), and if it's above a certain threshold, adopt "remove and re-add", while if it's below that threshold, keep the shift implementation.

Shifting was also the ObjC implementation, so for now, I'm keeping it like that, and we can revisit adoption of a new implementation later.